### ISSUE TITLE
"nbsphinx 0.4.1 requires sphinx>=1.6, which is not installed", when updating a virtualenv

### ISSUE THREAD

**Issue by jfbu** [spatialaudio/nbsphinx#278]
Well, I am sure this has some obvious explanation and is definitely not a nbsphinx issue...

I updated Pygments in a virtualenv:

```
$ pip install -U Pygments
Collecting Pygments
  Using cached https://files.pythonhosted.org/packages/13/e5/6d710c9cf96c31ac82657bcfb441df328b22df8564d58d0c4cd62612674c/Pygments-2.3.1-py2.py3-none-any.whl
nbsphinx 0.4.1 requires sphinx>=1.6, which is not installed.
Installing collected packages: Pygments
  Found existing installation: Pygments 2.2.0
    Uninstalling Pygments-2.2.0:
      Successfully uninstalled Pygments-2.2.0
Successfully installed Pygments-2.3.1
(sphinxvenv) (base)
```

The message `nbsphinx 0.4.1 requires sphinx>=1.6, which is not installed.` pops up. In fact I do have Sphinx installed in this virtual env, but via `pip install -e <sphinxrepodir>`.

**Comment by mgeier** [comment]
Hmmm, that's strange. I think this should also work with "editable" installs.

**Comment by jfbu** [comment]
I recalled only now that also nbsphinx was installed in an "editable" way in this environment, and my current checkout of the repo is at commit 9636d5338d8e.

```
$ pip show nbsphinx
Name: nbsphinx
Version: 0.4.1
Summary: Jupyter Notebook Tools for Sphinx
Home-page: UNKNOWN
Author: Matthias Geier
Author-email: Matthias.Geier@gmail.com
License: MIT
Location: /Users/xxxxxxxxxx/nbsphinx/src
Requires: docutils, jinja2, nbconvert, traitlets, nbformat, sphinx
Required-by: 
```

**Comment by mgeier** [comment]
Sorry, I have no clue what's going wrong here.

But as you said, it doesn't look like an `nbsphinx` issue ...

How did you install `pip`?

I think I had similar issues when Debian changed their `pip` package from installing to `/usr/local/lib/...` to installing to `~/.local/lib/...`.

I am skeptical about using plain `pip` because I never know which version that happens to be. I normally use `python3 -m pip install ...`.

Did you update your Python recently?
I had to re-install all my local pip-installed packages after a recent update to Python 3.7.

**Comment by jfbu** [comment]
> [...]
> 
> How did you install `pip`?

At root I have on my mac os x a Conda installation of Python. The I used ``virtualenv``. The `pip` is the one from that virtualenv. I upgraded it once or twice it is now at 18.1

```
$ pip --version
pip 18.1 from /Users/xxxxxxxxxxxx/sphinxvenv/lib/python3.6/site-packages/pip (python 3.6)
```

> 
> [...]
> 
> I am skeptical about using plain `pip` because I never know which version that happens to be. I normally use `python3 -m pip install ...`.
> 

Not used to that, but maybe I should.

> Did you update your Python recently?
> I had to re-install all my local pip-installed packages after a recent update to Python 3.7.

Yes I did update Python in my Conda base environment, but, I do think I then recreated the virtualenv from scratch after that update. At least I have done that once recently. Now, maybe I updated my base Python without following up in the sphinx dedicated virtualenv.

I will do it all again from scratch and report.

**Comment by jfbu** [comment]
I can not reproduce currently.

1. virtualenv sphinxvenv

2. activate it

3. pip install -e . from checkout of 1.8 branch of sphinx repo

4. go to nbsphinx local clone at current tip, then do `python3 setup.py develop`

5. Try to install a few things with pip (I installed sphinxcontrib_bibtex) and see if a message of nbsphinx pops up. None yet.

6. I also forcefully installed Pygments 2.2.0, and the upgraded it to 2.3.1 hoping to recreated original context. 

By the way, my Python is now at 3.6.8.

I have tried again (using `pip install -e .` for nbsphinx too) and do not manage to recreate the problem. I guess my virtualenv had gotten broken for some reason. I tested with Sphinx both at 1.8 and master and could not reproduce issue (which is probably not nbsphinx related as you said).

Sorry for annoyance, I will close this as I can't reproduce.
